src/libstd/sync/rwlock.rs

```diff
@@ -171,14 +171,16 @@ impl<T: ?Sized> RwLock<T> {
         RwLockReadGuard::new(&*self.inner, &self.data)
     }
 
-    /// Attempts to acquire this lock with shared read access.
+    /// Attempts to acquire this rwlock with shared read access.
+    ///
+    /// If the access could not be granted at this time, then `Err` is returned.
+    /// Otherwise, an RAII guard is returned which will release the shared access
+    /// when it is dropped.
```

Comment:
Hm ok so upon re-reading my original though that this indicates `Err` means the lock wasn't grabbed isn't quite right, it just means that if it couldn't be grabbed `Err` is returned, so looks good to me!